codes/C++/Reversing a linked list

```diff
@@ -0,0 +1,79 @@
+// Iterative C++ program to reverse
+// a linked list
+#include <iostream>
+using namespace std;
+ 
+/* Link list node */
+struct Node {
+    int data;
+    struct Node* next;
+    Node(int data)
+    {
+        this->data = data;
+        next = NULL;
+    }
+};
+ 
+struct LinkedList {
+    Node* head;
+    LinkedList() { head = NULL; }
+ 
+    /* Function to reverse the linked list */
+    void reverse()
+    {
+        // Initialize current, previous and
+        // next pointers
+        Node* current = head;
+        Node *prev = NULL, *next = NULL;
+ 
+        while (current != NULL) {
+            // Store next
+            next = current->next;
+ 
+            // Reverse current node's pointer
+            current->next = prev;
+ 
+            // Move pointers one position ahead.
+            prev = current;
+            current = next;
+        }
+        head = prev;
+    }
+ 
+    /* Function to print linked list */
+    void print()
+    {
+        struct Node* temp = head;
+        while (temp != NULL) {
+            cout << temp->data << " ";
+            temp = temp->next;
+        }
+    }
+ 
+    void push(int data)
+    {
+        Node* temp = new Node(data);
```

Comment:
Declaring a dynamic memory block in C/C++ with the `new` keyword must be freed somewhere when the program ends or when the class's destructor is called with the `delete` keyword otherwise it would cause a memory leak.